### ISSUE TITLE
[apex] False positive: ApexUnitTestMethodShouldHaveIsTestAnnotationRule should not flag methods with parameters 

### ISSUE THREAD

**Issue by catalandres** [pmd/pmd#2209]
**Affects PMD Version:**
(unable to determine)

**Rule:**
ApexUnitTestMethodShouldHaveIsTestAnnotationRule

**Description:**
It is common to factor out code from test methods into an auxiliary method that accepts parameters and assists in running the test. PMD is flagging those methods recommending that the `@isTest` flag be added. However, methods with that flag cannot have parameters.

**Code Sample demonstrating the issue:**

```apex
@isTest(isParallel=true)
public without sharing class DummyTest {
    @isTest 
    public static void testTrue() {
        auxiliaryMethod(1, 1, true);
    }
    @isTest 
    public static void testFalse() {
        auxiliaryMethod(1, 2, false);
    }
    // This method should not be required to be flagged with @isTest
    public static void auxiliaryMethod(Integer a, Integer b, Boolean expected) {
        System.assertEquals(expected, a == b);
    }
}
```

**Running PMD through:** *VS Code + chuckjonas.apex-pmd plugin*

**Comment by rsoesemann** [comment]
You have to live with that. Maybe use @supresswarnings. Static code analysers just are not smart enough to understand such cases.

**Comment by oowekyala** [comment]
@rsoesemann if methods with parameters cannot have the annotation then i don't see what's complicated here? Do I misunderstand the problem?
